mm/shm/shmdt.c

```diff
@@ -116,8 +115,7 @@ int shmdt(FAR const void *shmaddr)
 
   /* Free the virtual address space */
 
-  gran_free(group->tg_shm.gs_handle, (FAR void *)shmaddr,
-            region->sr_ds.shm_segsz);
+  shm_free(group, (FAR void *)shmaddr, region->sr_ds.shm_segsz);
```

Comment:
remove the cast

Reply:
Cannot, they are not of the same type. shmaddr is FAR const void * while shm_free eats FAR void *